server/controllers/finance/patientInvoice.js

```diff
@@ -358,7 +373,36 @@ function search(req, res, next) {
     req.query.uuid = db.bid(req.query.uuid);
   }
 
-  let queryObject = util.queryCondtion(sql, req.query);
+  if(req.query.reference){
+    additionalTokenQuery.push(' CONCAT(project.abbr, invoice.reference) = ?');
+    additionalTokenCondition = additionalTokenCondition.concat(req.query.reference);
+    delete req.query.reference;
+  }
+
+  if(req.query.billingDateFrom && req.query.billingDateTo){
+
+      additionalTokenQuery.push(' DATE(invoice.date) >= DATE(?)');
+      additionalTokenCondition = additionalTokenCondition.concat(req.query.billingDateFrom);
+      delete req.query.billingDateFrom;
+
+      additionalTokenQuery.push(' DATE(invoice.date) <= DATE(?)');
+      additionalTokenCondition = additionalTokenCondition.concat(req.query.billingDateTo);
+      delete req.query.billingDateTo;
+  }
+
+  if(req.query.is_distributable && req.query.is_distributable === 'all'){
```

Comment:
`is_distributable = 'all'` is the same as not filtering on `is_distributable`, correct?  Why have this option?